[DOC] Convert monitor, troubleshoot, set up test app to Alloy (#4026)
grafana/tempo: docs/sources/tempo/operations/monitor/set-up-monitoring.md

To set up monitoring, you need to:

> * Use Grafana Alloy collector to remote-write to Tempo and set up Grafana to visualize the tracing data by following [Set up a test app](https://grafana.com/docs/tempo/<TEMPO_VERSION>/setup/set-up-test-app/).

Reviewer: I'd either use `the Grafana Alloy collector` or drop the word `collector` here.

Author: ```suggestion
* Use Grafana Alloy to remote-write to Tempo and set up Grafana to visualize the tracing data by following [Set up a test app](https://grafana.com/docs/tempo/<TEMPO_VERSION>/setup/set-up-test-app/).
```

> * Update your collector configuration to scrape metrics to monitor for your Tempo data.

Reviewer: Are you referring to collectors in general? Or just to Alloy? If it's juts Alloy, then we could name it here instead of using `collector`.

Author: Will do. This tarted off as a doc to use both Agent Flow and Alloy. We changed it to just Alloy. This is a good catch. 
```suggestion
* Update your Alloy configuration to scrape metrics to monitor for your Tempo data.
```

This procedure assumes that you have set up Tempo [using the Helm chart](https://grafana.com/docs/tempo/<TEMPO_VERSION>/setup/helm-chart/) with [Grafana Alloy](https://grafana.com/docs/alloy/<TEMPO_VERSION>/set-up/install/).

> The steps outlined below use the Grafana Alloy configurations described in [Set up a test application for a Tempo cluster](https://grafana.com/docs/tempo/<TEMPO_VERSION>/setup/set-up-test-app/).

Reviewer: We've been defining `Grafana Alloy` in the metadata descriptions, H1s and at the first mention. Then all other mentions are just `Alloy`. I see a few places here where it's full name and short name, example line 69 and again in lines 87/88.  There are more like this.  I'm guessing it's an artefact of converting Grafana Agent to Grafana Alloy and then converting Flow to just Alloy.

It's not a fixed rule, just mentioning it. It's good with the full name too.

Author: You're exactly right that this is a left over from the conversion from Agent to Alloy. I'll merge your suggestions and then do a review of the doc for consistently.

Author: ```suggestion
The steps outlined below use the Alloy configurations described in [Set up a test application for a Tempo cluster](https://grafana.com/docs/tempo/<TEMPO_VERSION>/setup/set-up-test-app/).
```

In your Grafana instance, you'll need:

* [A Tempo data source](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/datasources/tempo/configure-tempo-data-source/) (created in the previous section)
* A [Mimir (Prometheus) data source](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/datasources/prometheus/)

## Enable Tempo metrics scraping

Tempo exposes Prometheus metrics from all of its components to allow meta-monitoring.
Grafana Alloy can both collect traces and act as a Prometheus scraper to retrieve metrics from Tempo. To use this capability, you need to configure Alloy to scrape from all of the components.

> Grafana Alloy lets you discover targets to scrape in a cluster via a variety of ways.

Author: ```suggestion
Alloy lets you discover targets to scrape in a cluster via a variety of ways.
```

> Usually for Prometheus metrics scraping, you would annotate the pods, services, and others, to signal that Grafana Alloy should scrape metrics from those objects using annotations such as `prometheus/scrape: true` as well as a port and path.

Author: ```suggestion
Usually for Prometheus metrics scraping, you would annotate the pods, services, and others, to signal that Alloy should scrape metrics from those objects using annotations such as `prometheus/scrape: true` as well as a port and path.
```

This lets you create a configuration that scrapes metrics from Tempo components and writes the data to a Mimir instance of your choice.

> Here’s a complete configuration for a [Grafana Alloy Helm](https://grafana.com/docs/alloy/<ALLOY_VERSION>/configure/kubernetes/) values file for Alloy, to scrape a running instance of Tempo.

Reviewer: This one converted to Alloy a bit oddly.

Author: I'm removing some of the 
```suggestion
This example provides a Helm values file for a [Grafana Alloy deployed on Kubernetes](https://grafana.com/docs/alloy/<ALLOY_VERSION>/configure/kubernetes/).
```

      rule {
        source_labels = ["__meta_kubernetes_pod_label_app_kubernetes_io_instance"]
        action = "replace"
        regex = "(.*)"
        replacement = "$1"
        target_label = "cluster"
      }
    }

    // Relabel data from Prometheus scraping.
    prometheus.relabel "tempo" {
      // Replace the existing job label with one comprised of the namespace and component.
      rule {
        source_labels = ["namespace", "k8s_component_name"]
        action = "replace"
        regex = "(.*?);(.*?)"
        replacement = "$1/$2"
        target_label = "job"
      }

      // Send the metrics to the Prometheus remote write component.
      forward_to = [prometheus.remote_write.tempo.receiver]
    }

    // Remote write the metrics to a Prometheus compatible endpoint (in this case Mimir).
    prometheus.remote_write "tempo" {
      endpoint {
        url = "https://mimir-cluster.distributor.mimir.svc.cluster.local:9001/api/v1/push"
      }
    }
```

This example doesn’t include ingestion for any other data such as traces for sending to Tempo, but can be included with some configuration updates.

> Refer to [Configure Grafana Alloy to remote-write to Tempo](https://grafana.com/docs/tempo/<TEMPO_VERSION>/setup/set-up-test-app/) for more information.

Author: ```suggestion
Refer to [Configure Alloy to remote-write to Tempo](https://grafana.com/docs/tempo/<TEMPO_VERSION>/setup/set-up-test-app/) for more information.
```
